dealii-policy-compatibility.md

```diff
@@ -0,0 +1,43 @@
+|**M4.** Give best effort at portability to key architectures (standard Linux distributions, GNU, Clang, vendor compilers, and target machines at ALCF, NERSC, OLCF). |Full| We strive to build on all platforms, and test regularly with GNU, Clang, and other compilers. We consider the inability of building on a platform as a bug. |
+|**M5.** Provide a documented, reliable way to contact the development team. |Full| Mailing list addresses are provided at http://www.dealii.org . |
+|**M6.** Respect system resources and settings made by other previously called packages (e.g. signal handling). |Full| Yes. The only exception that comes to mind is that deal.II enables traps/exceptions when arithmetic operations encounter signaling NaNs. We use signaling NaNs in deal.II to default initialize objects with invalid values to ensure they cannot be used before receiving proper values.|
+|**M7.** Come with an open source (BSD style) license. |Full| LGPL. |
+|**M8.** Provide a runtime API to return the current version number of the software. |Full| Via `Utilities::dealii_version_string()`. |
+|**M9.** Use a limited and well-defined symbol, macro, library, and include file name space. | Mostly| All header files are prefixed with `deal.II/`. All C++ names are in a namespace `dealii::`. (The exception of a few overloads of `std::pow`, `std::sin`, etc with arguments that are classes in namespace `dealii` -- consequently, these overloads cannot clash with other libraries.) Almost all macros are prefixed with `DEAL_II_`. The exception are a few macros for exception-related things; namely, `Assert`, `AssertThrow`, `AssertNothrow`, and a set of macros with names `DeclException*`. |
+|**M10.** Provide an xSDK team accessible repository (not necessarily publicly available). |Full| https://github.com/dealii/dealii |
+|**M11.** Have no hardwired print or IO statements that cannot be turned off. |Full| Yes. All logging is off by default. |
+|**M12.** For external dependencies, allow installing, building, and linking against an outside copy of external software. |Full| Yes. This is true for all dependencies. We bundle BOOST, TBB, and UMFPACK, but cmake will use an external version if one can be found. |
+|**M13.** Install headers and libraries under \<prefix\>/include and \<prefix\>/lib. |Full| Yes. The install path is chosen via `CMAKE_INSTALL_PREFIX`. |
+|**M14.** Be buildable using 64 bit pointers. 32 bit is optional. |Full| We make no assumptions. Whatever the compiler wants to use. |
+|**M15.** All xSDK compatibility changes should be sustainable. |Full| Yes. Rather than providing the xSDK options as add ons, we have incorporated them fully into the basic deal.II infrastructure, so that xSDK options get the same maintaince and support as any other feature. |
+|**M16.** The package must support production-quality installation compatible with the xSDK install tool and xSDK metapackage. |Full| Yes. deal.II is supported by SPACK. |
```

Comment:
since this mentions `xSDK metapackage`, i guess we need to add `deal.II` to https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/xsdk/package.py

If that's the case, I can open a PR for that.

Reply:
Yes, go ahead. We may be getting ahead of ourselves, but I see no reason not to open such a PR already at this time even while we're figuring other things out.

Reply:
@bangerth 

> Yes, go ahead. We may be getting ahead of ourselves, but I see no reason not to open such a PR already at this time even while we're figuring other things out.

see https://github.com/spack/spack/pull/7529